spec/integration/post_index_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
+
+RSpec.describe 'Post Index', type: :system do
+  before do
+    driven_by(:rack_test)
+  end
+
+  let(:user) do
+    User.create(name: 'Adnan', photo: 'https://unsplash.com/photos/ABC123', posts_counter: 5)
+  end
+
+  let!(:post) do
+    Post.create(title: 'My First Post', text: 'This is my first post', author: user, comments_counter: 0,
+                likes_counter: 0)
+  end
+
+  let!(:comment) do
+    Comment.create(text: 'This is my first comment', author: user, post:)
+  end
+
+  it "displays the user's profile picture on user posts page" do
+    visit users_path(user_id: user.id)
+    expect(page).to have_css("img[src='#{user.photo}']")
+  end
+
+  it "displays the user's name on user posts page" do
+    visit users_path(user_id: user.id)
+    expect(page).to have_content(user.name)
+  end
+
+  it "displays the user's post count on user posts page" do
+    visit users_path(user_id: user.id)
+    expect(page).to have_content(user.posts_counter)
+  end
+
+  it 'displays a post title on user posts page' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(post.title)
+  end
+
+  it 'displays a post text on user posts page' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(post.text)
+  end
+
+  it 'displays the first comments on a post' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(comment.text)
+  end
+
+  it 'displays the number of comments on a post' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(post.comments_counter)
+  end
+
+  it 'displays the number of likes on a post' do
+    visit user_posts_path(user_id: user.id)
+    expect(page).to have_content(post.likes_counter)
+  end
+
```

Comment:
![image](https://github.com/Cisco-the-wayword/BlogApp/assets/93039359/4dc68f99-9ddd-4f51-83ed-20d1213714e8)

- [ ] It would be nice if you double-check the project requirements and add the missing test case in your post_index_spec to complete the testing of the post_index page 😊